File: Resolução em python/ex2.py
def calc(preco, forma):
  if forma == 1:
    return preco * 0.85  # 15% de desconto à vista
  elif forma == 2:
    return preco * 0.9  # 10% de desconto no cartão de débito
  elif forma == 3:
    return preco * 0.95  # 5% de desconto no cartão de crédito
  else:
    return preco


preco_total = float(input("Informe o preço total da venda: "))
forma = int(
    input(
        "Informe a forma de pagamento (1 - à vista, 2 - cartão de débito, 3 - cartão de crédito): "
    ))

preco_final = calc(preco_total, forma)

print("O valor final a ser pago é: R$ {:.2f}".format(preco_final))

File: .ccls-cache/@home@runner@ListaJess-C/ex2.c
#include <stdio.h>

int ex2() {
  float preco_total;
  int forma_pagamento;
  float desconto = 0;
  float preco_final;

  printf("Informe o preço total da venda: ");
  scanf("%f", &preco_total);

  printf("Informe a forma de pagamento (1 - à vista, 2 - cartão de débito, 3 - "
         "cartão de crédito): ");
  scanf("%d", &forma_pagamento);

  switch (forma_pagamento) {
  case 1:
    desconto = preco_total * 0.15;
    break;
  case 2:
    desconto = preco_total * 0.10;
    break;
  case 3:
    desconto = preco_total * 0.05;
    break;
  default:
    printf("Forma de pagamento inválida.\n");
    break;
  }

  preco_final = preco_total - desconto;

  printf("Preço final a ser pago: %.2f\n", preco_final);

  return 0;
}